ddl/index.go

```diff
@@ -883,12 +886,79 @@ func (w *worker) addTableIndex(t table.Table, indexInfo *model.IndexInfo, reorgI
 	}
 	defer closeAddIndexWorkers(idxWorkers)
 
-	kvRanges, err := splitTableRanges(t, reorgInfo)
+	finish := false
+	for !finish {
+		kvRanges, err := splitTableRanges(t, reorgInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		err = w.buildKVRangesIndex(t, idxWorkers, kvRanges, job, reorgInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		finish, err = w.updateReorgInfo(t, reorgInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+// updateReorgInfo will find the next partition according to current reorgInfo.
+func (w *worker) updateReorgInfo(t table.Table, reorg *reorgInfo) (bool, error) {
+	pi := t.Meta().GetPartitionInfo()
+	if pi == nil {
+		return true, nil
+	}
+
+	pid, err := findNextPartitionID(reorg.PartitionID(), pi.Definitions)
 	if err != nil {
-		return errors.Trace(err)
+		// Fatal error, should not run here.
+		return false, errors.Errorf("wrong reorgInfo, partition id %d not found", reorg.PartitionID())
+	}
+	if pid == -1 {
+		return true, nil
+	}
+
+	start, end, err := getPartitionRange(reorg.d, t.Meta(), pid, reorg.Job.SnapshotVer)
+	if err != nil {
+		return false, errors.Trace(err)
 	}
 
-	return w.buildKVRangesIndex(t, idxWorkers, kvRanges, job, reorgInfo)
+	reorgMeta := model.NewDDLReorgMeta()
+	reorgMeta.PartitionID = pid
+	reorgMeta.EndHandle = end
+	reorg.Job.ReorgMeta = reorgMeta
+	reorg.StartHandle = start
+
+	log.Infof("[ddl] job %v update reorgInfo partitionID:%v, startHandle:%v, endHandle:%v", reorg.Job.ID, pid, start, end)
```

Comment:
I hope we can unify these logs. It's good to trace the bug.
s/[ddl]/[ddl-reorg]
s/`partitionID:%v, startHandle:%v, endHandle:%v"`/` partition %d range [%d, %d]`